### PR TITLE
config: Fix NetworkTopology breakage

### PR DESCRIPTION
**Pull request by erikd**
@karknu I was anot able to get to the bottom of what broke with my previous change so i have changed it back to having the `Maybe` inside the newtype. If nothing else I at least added some tests.

### CODE REVIEW

File: cardano-config/src/Cardano/Config/Parsers.hs [hunk 1]
```diff
@@ -232,15 +232,15 @@ parseNodeId desc =
...
 
 parseHostAddr :: Parser NodeHostAddress
 parseHostAddr =
-    option (NodeHostAddress <$> auto) (
+    option (eitherReader parseNodeHostAddress) (
           long "host-addr"
        <> metavar "HOST-NAME"
        <> help "Optionally limit node to one ipv6 or ipv4 address"
-       <> value (NodeHostAddress "127.0.0.1")
+       -- <> value Nothing
```
Comment (karknu) [inline cardano-config/src/Cardano/Config/Parsers.hs:243]: Should be ` <> (value $ NodeHostAddress Nothing)`
Without any specif address specified `cardano-node` will use any IPv4 and/or IPv6 address.

Comment (erikd) [reply cardano-config/src/Cardano/Config/Parsers.hs:243]: Fixed!

File: cardano-config/test/Test/Cardano/Config/Json.hs [hunk 2]
```diff
@@ -18,13 +20,17 @@ prop_roundtrip_NodeAddress_JSON =
...
+    Hedgehog.cover 1 "  has address" $ isJust (unNodeHostAddress $ naHostAddress na)
+    Hedgehog.cover 1 "  no address" $ isNothing (unNodeHostAddress $ naHostAddress na)
 
 prop_roundtrip_NodeHostAddress_JSON :: Property
 prop_roundtrip_NodeHostAddress_JSON =
   Hedgehog.property $ do
     nha <- Hedgehog.forAll genNodeHostAddress
     Hedgehog.tripping nha toJSON fromJSON
     Hedgehog.tripping nha encode decode
+    Hedgehog.cover 1 "  has address" $ isJust (unNodeHostAddress nha)
```
Comment (Jimbo4350) [inline cardano-config/test/Test/Cardano/Config/Json.hs:32]: :+1:

**Pull request merged**